### ISSUE TITLE
Implement smooth algorithm

### ISSUE THREAD

**Issue by brandonLi8** [phetsims/calculus-grapher#22]
Implement the smooth algorithm in the smooth method of the `OriginalCurve`.

I'm going to start by trying to understand the flash implementation.

**Comment by amanda-phet** [comment]
Just wanted to be sure you saw the design doc where Mike said "Smooth will replace every point with the average of its neighbors." 

I'm sure there's more to it than that, but just wanted to make sure you knew at least that part!

**Comment by brandonLi8** [comment]
Thanks @amanda-phet. I did some research into general [smoothing](https://en.wikipedia.org/wiki/Smoothing) algorithms and compared it with the flash code segment for smoothing:

```

    public function simpleSmoothAllPoints():void {
        var range:int = this.smoothingRange;
        var sumOfYs:Number;
        var N:Number;
        var dist:int;
        for ( var i:int = 0; i < this.nbrPoints; i++ ) {
            sumOfYs = 0;
            N = 0;
            for ( var j:int = -range; j <= range; j++ ) {
                //dist = Math.abs(j);
                if ( (i + j >= 0) && (i + j < this.nbrPoints) ) {
                    sumOfYs += this.y_arr[i + j]; //1/(dist+1)*this.y_arr[i+j];
                    N += 1;//1/(dist+1);
                }
                else {
                    N += 1;
                    if ( i + j < 0 ) {
                        sumOfYs += this.y_arr[0];
                    }
                    else {   //if(i+j >= this.nbrPoints)
                        sumOfYs += this.y_arr[this.nbrPoints - 1];
                    }
                }
            }//end of for j
            this.scratch_arr[i] = sumOfYs / N;
        }//end of for i
        for ( var k:int = 0; k < this.nbrPoints; k++ ) {
            this.y_arr[k] = this.scratch_arr[k];
        }
        this.takeDerivative();
        this.takeIntegral();
        this.updateViews();
    }//end of simpleSmoothAllPoints()
```


It looks like the flash sim is using the moving average algorithm described in https://en.wikipedia.org/wiki/Moving_average#Simple_moving_average. Doing this now.

**Comment by brandonLi8** [comment]
I finished implementing the smoothing algorithm in the commits above.


I played around with the smoothing algorithm locally and it looks good to my eye:

<details><summary>video:</summary>

![*](https://user-images.githubusercontent.com/42391580/86974334-61693880-c133-11ea-86aa-5b62ff3c757f.gif)

</details>


I also added the ?smoothingWindowWidth (Default 1.2) query parameter to adjust the "window" width of the smoothing algorithm. It isn't clear what the flash sim used so we might need to play around with it in the future. It's also good to know about the ?pointsPerCoordinate query parameter (which adjusts how close together the curve partition is).
